src/network/bmproto.py

```diff
@@ -461,7 +455,9 @@ def bm_command_addr(self):
                                 "rating": 0,
                                 "self": False,
                             }
-                addrQueue.put((stream, peer, self.destination))
+                    # since we don't track peers outside of knownnodes,
+                    # only spread if in knownnodes to prevent flood
+                    addrQueue.put((stream, peer, self.destination))
```

Comment:
`seenTime` ): it will update the node, maybe:
```patch
diff --git a/src/network/addrthread.py b/src/network/addrthread.py
index fa7a72c6..3f88f9ee 100644
--- a/src/network/addrthread.py
+++ b/src/network/addrthread.py
@@ -32,12 +32,12 @@ class AddrThread(StoppableThread):
                 for i in connections:
                     randomshuffle(chunk)
                     filtered = []
-                    for stream, peer, source in chunk:
+                    for stream, peer, source, seen in chunk:
                         if i.destination in (peer, source):
                             continue
                         if stream not in i.streams:
                             continue
-                        filtered.append((stream, peer, time()))
+                        filtered.append((stream, peer, seen))
                     if filtered:
                         self.logger.warning(
                             'sending addr of len %s', len(filtered))
diff --git a/src/network/bmproto.py b/src/network/bmproto.py
index 0915a5ec..b496e2b0 100644
--- a/src/network/bmproto.py
+++ b/src/network/bmproto.py
@@ -457,7 +457,7 @@ class BMProto(AdvancedDispatcher, ObjectTracker):
                             }
                     # since we don't track peers outside of knownnodes,
                     # only spread if in knownnodes to prevent flood
-                    addrQueue.put((stream, peer, self.destination))
+                    addrQueue.put((stream, peer, self.destination, seenTime))
         return True
 
     def bm_command_portcheck(self):
```

Reply:
yes